name: uds
channels:
  - conda-forge
  - defaults
dependencies:
  - defaults::python==3.9.16    # avoids jupyterlab error on Windows
  - defaults::notebook
  - defaults::jupyterlab
  - defaults::pip
  - cenpy
  - numpy==1.23.5    # version is for partridge compatibility, until this issue is resolved:https://github.com/remix/partridge/issues/77 
  - pandas==1.5.3  # to avoid incompatibility with genesis in pandas 2.0.0
  - geopandas
  - contextily
  - requests
  - shapely
  - bs4
  - matplotlib
  - seaborn
  - PyPDF2
  - pdfminer.six
  - nltk
  - gensim
  - openpyxl
  - pyLDAvis
  - textblob
  - scikit-learn
  - pandasql
  - sqlalchemy==1.4.46 # version is for pandasql compatibility, until this issue is resolved: https://stackoverflow.com/questions/75315117/attributeerror-connection-object-has-no-attribute-connect-when-use-df-to-sq
  - praw
  - wordcloud
  - pip:
    - partridge
